tests/scalers_go/azure_queue_restore_original/azure_queue_restore_original_test.go

```diff
@@ -0,0 +1,195 @@
+//go:build e2e
```

Comment:
This e2e test doesn't check azure queue itself but scaling internals (the restore to original), 
Could you move to `internals` folder and remove the reference to the scaler (which isn't important in what is tested)?

Reply:
Should I also move the Trigger Auth one, or keep it in scalers?

Reply:
Good question, but I don't have a good answer because the line is really fuzzy 😢 
I'd keep them a part of the scaler in this case because this auth method is totally attached to the scaler itself (we are testing that we can connect recovering the correct secrets from the trigger). In case of tests where the important thing to check is how we recover the secrets from the upstream (vault, Key vault), I'd put them in `secret-provider` folder.
But again, this is totally opinionated and maybe we should remove the 3rd folder and keep only `scalers` and `internals`.
WDYT @v-shenoy @kedacore/keda-core-contributors ?

Reply:
I think having it under secret-provider makes more sense than under scalers.

Reply:
Would you add all the Trigger Auth tests under secret-provider? Even if they are not using any external provider for the secrets? 🤔
Don't you think that there will be a lot of e2e tests there testing similar things? I mean, all the scalers can use trigger authentication to recover secrets and the only changes are in the scalers code and not in the Trigger Authentication mechanics itself. That's why I'd split them based on what are they testing. 
But most probably I'm overthinking

Reply:
Not all tests that use trigger auth, but tests that are specifically testing the trigger auth functionality. These I would put into secret-providers, and the rest along with the scalers.

Reply:
that's true.
Do you think that current `trigger_auth` test should be `trigger_auth_secret` to specify that it's testing the integration with k8s secrets as secrets source? (not blocker, we can modify it once we are adding other sources)